mikeio1d/result_reader_writer/result_reader_copier.py

```diff
@@ -70,26 +95,55 @@ def create_data_frame(self, data_entries, column_names):
         data_pointer_net = IntPtr(data_pointer)
         self.result_data_copier.CopyData(data_pointer_net, data_entries)
 
-        df = pd.DataFrame(data_array, index=self.time_index, columns=column_names)
+        columns = self.create_column_index(timeseries_ids, column_mode=column_mode)
+
+        df = pd.DataFrame(data_array, index=self.time_index, columns=columns)
 
         self.update_time_quantities(df)
 
         return df
 
-    def get_all_data_entries_and_column_names(self):
+    def create_column_index(
```

Comment:
Is this also relevant for result_reader_query and should go to the parent class?